apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -671,7 +671,12 @@ func (dc *CassandraDatacenter) GetConfigAsJSON(config []byte) (string, error) {
 			return "", errors.Wrap(err, "Error parsing Spec.Config for CassandraDatacenter resource")
 		}
 
-		if err := modelParsed.Merge(configParsed); err != nil {
+		collisionFn := func(dest, src interface{}) interface{} {
```

Comment:
>This allows overwriting also the seeds

I could modify `collisionFn` to only allow the datacenter name change. This is really intended as use at your own risk and primarily for k8ssandra-operator.

>That means, label queries, logs etc will not find this given name here. I guess that's no problem if user understands it's the CassandraDatacenter.Name, not Cassandra's Datacenter name.

Good point. Here was my motivation. A user creates a K8ssandraCluster with two DCs - `dc1` and `dc2`. When https://github.com/k8ssandra/k8ssandra-operator/issues/615 is fixed, the actual CassandraDatacenter names will be something generated by k8ssandra-operator. When I run `nodetool status`, I would expect to see `dc1` and `dc2`. This might be confusing.

I guess we need to take a step back and think about this some more.